dvc/proc/process.py

```diff
@@ -0,0 +1,198 @@
+import json
+import logging
+import os
+import shlex
+import subprocess
+from contextlib import AbstractContextManager
+from dataclasses import dataclass
+from typing import List, Optional, TextIO, Union
+
+from funcy import cached_property
+from shortuuid import uuid
+
+from dvc.utils.fs import makedirs
+
+from .exceptions import TimeoutExpired
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ProcessInfo:
+    PARAM_PID = "pid"
+    PARAM_STDIN = "stdin"
+    PARAM_STDOUT = "stdout"
+    PARAM_STDERR = "stderr"
+    PARAM_RETURNCODE = "returncode"
+
+    pid: int
+    stdin: Optional[str]
+    stdout: Optional[str]
+    stderr: Optional[str]
+    returncode: Optional[int]
+
+    @classmethod
+    def from_dict(cls, d):
+        return cls(
+            d.get(cls.PARAM_PID),
+            d.get(cls.PARAM_STDIN),
+            d.get(cls.PARAM_STDOUT),
+            d.get(cls.PARAM_STDERR),
+            d.get(cls.PARAM_RETURNCODE),
+        )
+
+    def to_dict(self):
```

Comment:
There is [`dataclasses.asdict`](https://docs.python.org/3/library/dataclasses.html) that does this already.